cmd/tree-walk.go

```diff
@@ -29,29 +29,20 @@ type TreeWalkResult struct {
 }
 
 // Return entries that have prefix prefixEntry.
-// Note: input entries are expected to be sorted.
+// The supplied entries are modified and the returned string is a subslice of entries.
 func filterMatchingPrefix(entries []string, prefixEntry string) []string {
-	start := 0
-	end := len(entries)
-	for {
-		if start == end {
-			break
-		}
-		if HasPrefix(entries[start], prefixEntry) {
-			break
-		}
-		start++
+	if len(entries) == 0 || prefixEntry == "" {
+		return entries
 	}
-	for {
-		if start == end {
-			break
-		}
-		if HasPrefix(entries[end-1], prefixEntry) {
-			break
+	// Write to the beginning of entries.
+	dst := entries[:0]
+	for _, s := range entries {
+		if !strings.HasPrefix(s, prefixEntry) {
```

Comment:
HasPrefix is used for windows @klauspost to allow case insensitive prefix.

Reply:
Ah, so prefix is S3 prefix - actually I still expected Windows to be case sensitive. But hey.

Reply:
@harshavardhana Fixed.